components/ws-manager/pkg/manager/manager.go

```diff
@@ -290,6 +293,33 @@ func (m *Manager) StartWorkspace(ctx context.Context, req *api.StartWorkspaceReq
 		return nil, xerrors.Errorf("cannot create workspace pod: %w", err)
 	}
 
+	// remove annotation to signal that workspace pod was indeed created and scheduled on the node
+	// and use longer backoff to ensure we do remove this annotation
+	backoff = wait.Backoff{
+		Steps:    7,
+		Duration: 100 * time.Millisecond,
+		Factor:   2.0,
+		Jitter:   0.1,
+	}
+	err = retry.RetryOnConflict(backoff, func() error {
+		var tempPod corev1.Pod
+		err := m.Clientset.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}, &tempPod)
+		if err != nil {
+			clog.WithError(err).WithField("pod.Namespace", pod.Namespace).WithField("pod.Name", pod.Name).Error("was unable to get pod to remove annotation")
+			return err
+		}
+		delete(tempPod.Annotations, attemptingToCreatePodAnnotation)
+		err = m.Clientset.Update(ctx, &tempPod)
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
this should use `markWorkspace` instead which carries the concurrent modification retry logic